File: software/examples/rust/example_simple.rs
use std::{error::Error, io};

use tinkerforge::{air_quality_bricklet::*, ip_connection::IpConnection};

const HOST: &str = "localhost";
const PORT: u16 = 4223;
const UID: &str = "XYZ"; // Change XYZ to the UID of your Air Quality Bricklet.

fn main() -> Result<(), Box<dyn Error>> {
    let ipcon = IpConnection::new(); // Create IP connection.
    let aq = AirQualityBricklet::new(UID, &ipcon); // Create device object.

    ipcon.connect((HOST, PORT)).recv()??; // Connect to brickd.
                                          // Don't use device before ipcon is connected.

    // Get current all values.
    let all_values = aq.get_all_values().recv()?;

    println!("IAQ Index: {}", all_values.iaq_index);

    if all_values.iaq_index_accuracy == AIR_QUALITY_BRICKLET_ACCURACY_UNRELIABLE {
        println!("IAQ Index Accuracy: Unreliable");
    } else if all_values.iaq_index_accuracy == AIR_QUALITY_BRICKLET_ACCURACY_LOW {
        println!("IAQ Index Accuracy: Low");
    } else if all_values.iaq_index_accuracy == AIR_QUALITY_BRICKLET_ACCURACY_MEDIUM {
        println!("IAQ Index Accuracy: Medium");
    } else if all_values.iaq_index_accuracy == AIR_QUALITY_BRICKLET_ACCURACY_HIGH {
        println!("IAQ Index Accuracy: High");
    }

    println!("Temperature: {} °C", all_values.temperature as f32 / 100.0);
    println!("Humidity: {} %RH", all_values.humidity as f32 / 100.0);
    println!("Air Pressure: {} mbar", all_values.air_pressure as f32 / 100.0);

    println!("Press enter to exit.");
    let mut _input = String::new();
    io::stdin().read_line(&mut _input)?;
    ipcon.disconnect();
    Ok(())
}

File: software/examples/vbnet/ExampleSimple.vb
Imports System
Imports Tinkerforge

Module ExampleSimple
    Const HOST As String = "localhost"
    Const PORT As Integer = 4223
    Const UID As String = "XYZ" ' Change XYZ to the UID of your Air Quality Bricklet

    Sub Main()
        Dim ipcon As New IPConnection() ' Create IP connection
        Dim aq As New BrickletAirQuality(UID, ipcon) ' Create device object

        ipcon.Connect(HOST, PORT) ' Connect to brickd
        ' Don't use device before ipcon is connected

        ' Get current all values
        Dim iaqIndex, temperature, humidity, airPressure As Integer
        Dim iaqIndexAccuracy As Byte

        aq.GetAllValues(iaqIndex, iaqIndexAccuracy, temperature, humidity, airPressure)

        Console.WriteLine("IAQ Index: " + iaqIndex.ToString())

        If iaqIndexAccuracy = BrickletAirQuality.ACCURACY_UNRELIABLE Then
            Console.WriteLine("IAQ Index Accuracy: Unreliable")
        Else If iaqIndexAccuracy = BrickletAirQuality.ACCURACY_LOW Then
            Console.WriteLine("IAQ Index Accuracy: Low")
        Else If iaqIndexAccuracy = BrickletAirQuality.ACCURACY_MEDIUM Then
            Console.WriteLine("IAQ Index Accuracy: Medium")
        Else If iaqIndexAccuracy = BrickletAirQuality.ACCURACY_HIGH Then
            Console.WriteLine("IAQ Index Accuracy: High")
        End If

        Console.WriteLine("Temperature: " + (temperature/100.0).ToString() + " °C")
        Console.WriteLine("Humidity: " + (humidity/100.0).ToString() + " %RH")
        Console.WriteLine("Air Pressure: " + (airPressure/100.0).ToString() + " hPa")

        Console.WriteLine("Press key to exit")
        Console.ReadLine()
        ipcon.Disconnect()
    End Sub
End Module
